api/datastore/sql/migrations/old.go

```diff
@@ -0,0 +1,112 @@
+package migrations
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"github.com/fnproject/fn/api/common"
+	"github.com/jmoiron/sqlx"
+	"github.com/pressly/goose"
+)
+
+var oldMigrationsTable = "schema_migrations"
+
+func checkOldMigrationTableVersionIfExists(db *sqlx.DB) (version int64, dirty bool, err error) {
+	ctx := context.Background()
+
+	q := db.QueryRowContext(
+		ctx, "SELECT version, dirty FROM "+oldMigrationsTable+" LIMIT 1")
+	q.Scan(&version, &dirty)
+	if err == sql.ErrNoRows {
+		return -1, false, nil
+	} else if err != nil {
+		return -1, false, err
+	}
+	return version, dirty, nil
+}
+
+func checkOldMigration(ctx context.Context, db *sqlx.DB) (int64, error) {
+	log := common.Logger(ctx)
+	current, dirty, err := checkOldMigrationTableVersionIfExists(db)
+	if err != nil {
+		return -1, err
+	}
+	if dirty {
+		log.Fatal("database corrupted")
```

Comment:
"database corrupted, please return the database to a stable state under this migration version (run the down migration at the version indicated?) in table `schema_migrations` and unset the dirty bit in that table to restore the database to a working condition"

"database corrupted" is pretty jarring, and this _should_ be recoverable. it just means a migration was running and the db exited, should just have to run down migration and then unset the dirty bit.

Reply:
fixed